src/MinterGateway.sol

```diff
@@ -774,13 +774,13 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
 
         if (missedIntervals_ == 0) return;
 
-        // Save until when the minter has been penalized for missed intervals to prevent double penalizing them.
-        minterState_.penalizedUntilTimestamp = missedUntil_;
-
         uint112 penaltyPrincipal_ = _imposePenalty(minter_, uint152(principalOfActiveOwedM_) * missedIntervals_);
 
         if (penaltyPrincipal_ == 0) return;
 
+        // Save until when the minter has been penalized for missed intervals to prevent double penalizing them.
+        minterState_.penalizedUntilTimestamp = missedUntil_;
```

Comment:
`missedUntil_` is only different from `0` if `missedIntervals_` is different than `0` no?
I guess we can reach this path if `missedIntervals_` is greater than `0` but the calculation ends up rounding down `penaltyPrincipal_` to `0`.
Seems like a good change.